resources/web/wwi/protoVisualizer/Parameter.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+import {getAParameterId} from '../nodes/utils/id_provider.js';
+import {stringifyType} from './Vrml.js';
+
+export default class Parameter {
+  #id = getAParameterId();
+  #type;
+  #name;
+  #value;
+  #defaultValue;
+  #isTemplateRegenerator;
+  constructor(node, name, type, defaultValue, value, isTemplateRegenerator) {
+    this.node = node; // node this parameter belongs to
+    this.type = type;
+    this.name = name;
+    this.defaultValue = defaultValue;
+    this.value = value;
+    this.isTemplateRegenerator = isTemplateRegenerator;
+    this.parentNode = undefined;
+  }
+
+  get value() {
+    return this.#value;
+  }
+
+  set value(v) {
+    if (v.type() !== this.type)
+      throw new Error('Type mismatch, setting ' + stringifyType(v.type()) + ' to ' + stringifyType(this.type) + ' parameter.');
+
+    this.#value = v;
+  }
+
+  get defaultValue() {
+    return this.#defaultValue;
+  }
+
+  set defaultValue(v) {
+    if (v.type() !== this.type)
+      throw new Error('Type mismatch, setting ' + stringifyType(v.type()) + ' to ' + stringifyType(this.type) + ' parameter.');
+
+    this.#defaultValue = v;
+  }
+
+  get id() {
+    return this.#id;
+  }
+
+  set id(value) {
+    this.#id = value;
+  }
+
+  get type() {
+    return this.#type;
+  }
+
+  set type(type) {
+    this.#type = type;
+  }
+
+  get name() {
+    return this.#name;
+  }
+
+  set name(value) {
+    this.#name = value;
+  }
+
+  get isTemplateRegenerator() {
+    return this.#isTemplateRegenerator;
+  }
+
+  set isTemplateRegenerator(value) {
+    this.#isTemplateRegenerator = value;
+  }
+
+  setValueFromJavaScript(v) {
+    this.value.setValueFromJavaScript(v);
```

Comment:
Not sure if inside the class we should use the `getter` or directly access the attribute with `this.#value`. Personally I would tend to access directly the value.